src/stapi_fastapi/routers/product_router.py

```diff
@@ -63,9 +63,28 @@ def __init__(
             tags=["Products"],
         )
 
+        self.add_api_route(
+            path="/order-parameters",
+            endpoint=self.get_product_order_parameters,
+            name=f"{self.root_router.name}:{self.product.id}:get-order-parameters",
+            methods=["GET"],
+            summary="Get order parameters for the product",
+        )
+
+        async def _create_order(
+            payload: OrderRequest,
+            request: Request,
+            response: Response,
+        ) -> Order:
+            return await self.create_order(payload, request, response)
+
+        _create_order.__annotations__["payload"] = OrderRequest[
+            product.order_parameters  # type: ignore
+        ]
+
         self.add_api_route(
             path="/order",
-            endpoint=self.create_order,
+            endpoint=_create_order,
             name=f"{self.root_router.name}:{self.product.id}:create-order",
             methods=["POST"],
             response_class=GeoJSONResponse,
```

Comment:
Did you decide it wasn't worth specifying a more specifically-typed response model here? Like `response_model=Order[Geometry, self.product.constraints],`?

Reply:
I didn't because I got an error when I tried, but now that I feel more confident with this, I'll take another crack at it. 

Immediately, Mypy fails to validate it saying it doesn't know what `self.product.constraints` is, and the tests throw a validation error -- which could be a true positive for an error.

Reply:
added some annotations, and it seems to be working as intended -- one immediate problem is that instead of `Order[Geometry, self.product.constraints]`, the type is `Order[self.product.constraints, self.product.order_parameters]`